code/modules/medical/genetics/geneticsMachines.dm

```diff
@@ -1103,17 +1095,54 @@
 			"cooldown" = src.equipment[GENETICS_RECLAIMER] - world.time,
 		))
 
+
+	for(var/key in .)
+		var/json = json_encode(.[key])
+		var/msg = url_encode(json)
+		boutput(world,"[key]:[length(msg)]:[md5(msg)]")
+
+
+
 /obj/machinery/computer/genetics/ui_static_data(mob/user)
-	var/to_send = list()
+	. = list("bioEffects"=list(),
+					"research"=list(),
+					"boothCost" = genResearch.isResearched(/datum/geneticsResearchEntry/genebooth) ? genResearch.genebooth_cost : -1,
+					"injectorCost" = genResearch.isResearched(/datum/geneticsResearchEntry/injector) ? genResearch.injector_cost : -1,
+					"saveSlots" = genResearch.isResearched(/datum/geneticsResearchEntry/saver) ? genResearch.max_save_slots : 0,
+					"precisionEmitter" = genResearch.isResearched(/datum/geneticsResearchEntry/rad_precision),
+					"materialMax" = genResearch.max_material,
+					"mutantRaces" = list(list(
+						"name" = "Human",
+						"icon" = "template",
+						"ref" = "\ref[null]",
```

Comment:
`"\ref[null]"` seems weird.

Reply:
It keeps the strings consistent for comparisons in js and ui_interaction when value is passed back to DM.

But I agreed, it does seem weird.